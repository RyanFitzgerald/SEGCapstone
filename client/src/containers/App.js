import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { Redirect } from 'react-router-dom';
import * as api from '../api';

// Import components
import Header from '../components/Header';
import Home from '../components/Home';
import Client from './Client';
import Project from './Project';
import User from './User';
import NoMatch from '../components/NoMatch';
import Login from './Login';

class App extends Component {
  constructor() {
    super();

    // Bind functions
    this.setActiveTab = this.setActiveTab.bind(this);
    this.isLoggedIn = this.isLoggedIn.bind(this);
    this.logout = this.logout.bind(this);
    this.getCurrentUser = this.getCurrentUser.bind(this);

    // Set state defaults
    this.state = {
      activeTab: 1,
      isLoggedIn: false,
      user: false
    };
  }

  setActiveTab(tab) {
    this.setState({activeTab: tab});
  }

  componentDidMount() {
    this.isLoggedIn();

    // TODO get user only when logged in
    this.getCurrentUser();
  }

  isLoggedIn() {
    api.isLoggedIn().then(result => {
      const isLoggedIn = result;
      this.setState({isLoggedIn});
    });
  }

  logout() {
    api.logout().then(
      window.location.reload()
    );
  }

  // Retrieve user information
  getCurrentUser() {
    api.getCurrentUser().then(result => {
        const user = result;
        this.setState({user});
    });
  }

  render() {
    if (this.state.isLoggedIn) {
      return (
        <div className="app">
          <Header user={this.state.user} logout={this.logout} activeTab={this.state.activeTab}/>
  
          <Switch>
            <Route exact path="/" render={() =>
              <Home setActiveTab={this.setActiveTab}/>
            } />
            <Route path="/clients" render={() =>
              <Client setActiveTab={this.setActiveTab}/>
            } />
            <Route path="/projects" render={() =>
              <Project setActiveTab={this.setActiveTab}/>
            } />
            <Route path="/users" render={() =>
              <User setActiveTab={this.setActiveTab}/>
            } />
            <Route component={NoMatch}/>
          </Switch>
        </div>
      );
    } else {
      return (
        <div className="app">
          <Login />
        </div>
      );  
    }  
  }
}

export default App;
