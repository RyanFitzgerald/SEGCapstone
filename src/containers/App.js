import React, { Component, PropTypes } from 'react';
import {Route, Switch} from 'react-router-dom';

// Import components
import Sidebar from '../components/Sidebar';
import Header from '../components/Header';
import Home from '../components/Home';
import NoMatch from '../components/NoMatch';
import Client from './Client';
import Project from './Project';

class App extends Component {
    constructor(props) {
        super(props);

        // Default States
        this.state = {
            activeTab: 1,
            title: 'Dashboard'
        };
    }

    setActiveTab = (tab) => {
        this.setState({
            activeTab: tab
        });
    }

    setHeaderTitle = (newTitle) => {
        this.setState({
            title: newTitle
        });
    }

    render() {
        return (
            <div className="app-container">
                <Sidebar activeTab={this.state.activeTab} />

                <div id="dashboard">
                    <Header title={this.state.title} />

                    <Switch>
                        <Route exact path="/" render={() =>
                            <Home setActiveTab={this.setActiveTab} setHeaderTitle={this.setHeaderTitle} />
                        } />
                        <Route path="/projects" render={() =>
                            <Project setActiveTab={this.setActiveTab} setHeaderTitle={this.setHeaderTitle} />
                        } />
                        <Route path="/clients" render={() =>
                            <Client setActiveTab={this.setActiveTab} setHeaderTitle={this.setHeaderTitle} />
                        } />
                        <Route component={NoMatch} />
                    </Switch>
                </div>
            </div>
        );
    }
}

App.propTypes = {
    children: PropTypes.node
};

export default App;
