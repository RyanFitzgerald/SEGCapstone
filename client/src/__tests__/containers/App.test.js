import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MemoryRouter, Route } from 'react-router-dom';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// Import components
import App from '../../Containers/App';

// Setup axios mock
const mock = new MockAdapter(axios);

describe('<App />', () => {
  it('should render correctly', () => {
    mock.onGet('/auth/isLoggedIn').reply(200, true);
    const output = shallow(
      <App/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

