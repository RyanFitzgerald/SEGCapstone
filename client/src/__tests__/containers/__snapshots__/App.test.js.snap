// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<div
  className="app"
>
  <ToastContainer
    autoClose={3000}
    bodyClassName=""
    className={null}
    closeButton={<DefaultCloseButton />}
    closeOnClick={true}
    hideProgressBar={true}
    newestOnTop={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName=""
    style={null}
    toastClassName=""
    transition={[Function]}
    type="success"
  />
  <Switch>
    <Route
      path="/login"
      render={[Function]}
    />
    <Route
      render={[Function]}
    />
  </Switch>
</div>
`;
